rust/connlib/tunnel/src/dns.rs

```diff
@@ -261,33 +298,32 @@ where
         "Developer error: we should be always be able to create a MessageBuilder from a Vec",
     );
 
-    let Some(resource) = resource else {
+    if records.is_empty() {
```

Comment:
I've looked this up recently and no records doesn't necessarily mean NXDOMAIN. It is a valid DNS answer to return an empty set of records if the name is known but doesn't have any records attached.

I don't think there is a case where connlib should make an NXDOMAIN response:

- If the query is for a resource, we know the name.
- If we don't know the name, we need to forward and thus the next resolver needs to handle this.